assets/app/scripts/controllers/deployment.js

```diff
@@ -73,6 +73,7 @@ angular.module('openshiftConsole')
             });
             activeDeployment = DeploymentsService.getActiveDeployment(deploymentsForConfig);
             $scope.isActive = activeDeployment && activeDeployment.metadata.uid === $scope.deployment.metadata.uid;
+            $scope.logCanRun = !(_.includes(['New', 'Pending'], $filter('deploymentStatus')(activeDeployment)));
```

Comment:
You can in theory view logs for older deployments as long as the deployer pod still exists. Usually it's gone, though.

It would be more accurate to get the pods and see if the one in the deployer pod annotation is there. I was considering getting the pods anyway so we can show the pod donut on the browse deployment page.